pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1226,13 +1226,11 @@ func altStatName(statPattern string, host string, subset string, dnsDomain strin
 }
 
 func shortHostName(host string, dnsDomain string) string {
-	shortHost := strings.TrimSuffix(host, dnsDomain)
-	if parts := strings.Split(dnsDomain, "."); len(parts) > 1 {
-		shortHost += parts[0] // k8s will have namespace.<domain>
-	} else {
-		shortHost = host
+	if index := strings.Index(dnsDomain, "."); index != -1 {
+		// Exclude the namespace from dnsDomain and trim the actual domain part from host.
```

Comment:
this is k8s specific logic isn't it? what happens if we end up having a VM dns here? IIRC, there is an attribute in the service for whether it is in k8s or not. We could use that to add this logic. WDYT?

Reply:
Sure. If it is there, I can add that additional check.

Reply:
Do you have it handy? I count't find it?

Reply:
https://github.com/istio/istio/blob/master/pilot/pkg/model/service.go#L355

Reply:
@rshriram Thinking about it a bit more, I think we should just look at host it it has k8sDomain and trim the domain part - for other VM hosts since there is no easy way to determine the domain, we should return as is. See if that makes sense. If there are any better ideas, let me know - will change this.